<?xml version='1.0' encoding='UTF-8'?>
<root version="2.3.8">
 <tabbed_widget name="Main Window" parent="main_window">
  <Tab containers="1" tab_name="tab1">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockSplitter sizes="0.5;0.5" orientation="|" count="2">
      <DockArea name="...">
       <plot mode="TimeSeries" style="Lines">
        <range left="5.315392" bottom="-0.138320" top="5.671102" right="20.712288"/>
        <limitY/>
        <curve color="#05ff00" name="/residuals_gt/cumulateError/data"/>
       </plot>
      </DockArea>
      <DockSplitter sizes="0.500602;0.499398" orientation="-" count="2">
       <DockArea name="...">
        <plot mode="XYPlot" style="Lines">
         <range left="-0.254163" bottom="-0.109703" top="0.056991" right="0.137751"/>
         <limitY/>
         <curve color="#ff7f0e" name="/residuals_gt/d[x/data;y/data]" curve_x="/residuals_gt/dx/data" curve_y="/residuals_gt/dy/data"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" style="Lines">
         <range left="5.315392" bottom="-0.495765" top="0.012092" right="20.712288"/>
         <limitY/>
         <curve color="#00e2ff" name="/residuals_gt/dtheta/data"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab2">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockSplitter sizes="0.5;0.5" orientation="|" count="2">
      <DockArea name="...">
       <plot mode="TimeSeries" style="Lines">
        <range left="4.519565" bottom="-0.270892" top="11.106558" right="20.742411"/>
        <limitY/>
        <curve color="#05ff00" name="/residuals_od/cumulateError/data"/>
       </plot>
      </DockArea>
      <DockSplitter sizes="0.500602;0.499398" orientation="-" count="2">
       <DockArea name="...">
        <plot mode="XYPlot" style="Lines">
         <range left="-0.151878" bottom="-0.034319" top="0.067749" right="0.088093"/>
         <limitY/>
         <curve color="#ff7f0e" name="/residuals_od/d[x/data;y/data]" curve_x="/residuals_od/dx/data" curve_y="/residuals_od/dy/data"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" style="Lines">
         <range left="4.519565" bottom="-0.104726" top="0.016892" right="20.742411"/>
         <limitY/>
         <curve color="#00e2ff" name="/residuals_od/dtheta/data"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab3">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockSplitter sizes="0.5;0.5" orientation="|" count="2">
      <DockArea name="...">
       <plot mode="XYPlot" style="Lines">
        <range left="-0.673509" bottom="-1.515609" top="0.245028" right="1.142222"/>
        <limitY/>
        <curve color="#ff5800" name="/our_odom/pose/position/[x;y]" curve_x="/our_odom/pose/position/x" curve_y="/our_odom/pose/position/y"/>
        <curve color="#00e2ff" name="/real_time_gt/position/[x;y]" curve_x="/real_time_gt/position/x" curve_y="/real_time_gt/position/y"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" style="Lines">
        <range left="0.000000" bottom="-8.999956" top="368.998209" right="20.759310"/>
        <limitY/>
        <curve color="#ff5800" name="our_theta_360"/>
        <curve color="#00e2ff" name="gt_theta_360"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab containers="1" tab_name="tab4">
   <Container>
    <DockSplitter sizes="1" orientation="-" count="1">
     <DockSplitter sizes="0.5;0.5" orientation="|" count="2">
      <DockArea name="...">
       <plot mode="XYPlot" style="Lines">
        <range left="-0.679132" bottom="-1.534254" top="0.037469" right="0.941774"/>
        <limitY/>
        <curve color="#ff5800" name="/our_odom/pose/position/[x;y]" curve_x="/our_odom/pose/position/x" curve_y="/our_odom/pose/position/y"/>
        <curve color="#05ff00" name="/real_time_od/pose/position/[x;y]" curve_x="/real_time_od/pose/position/x" curve_y="/real_time_od/pose/position/y"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" style="Lines">
        <range left="0.000000" bottom="-8.999999" top="368.999971" right="20.759310"/>
        <limitY/>
        <curve color="#ff5800" name="our_theta_360"/>
        <curve color="#05ff00" name="scout_theta_360"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="2"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis=""/>
  </plugin>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <use_renaming_rules value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="MQTT Subscriber"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <use_renaming_rules value="true"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="ROS /rosout Visualization" status="idle"/>
  <plugin ID="ROS Topic Re-Publisher" status="idle"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations>
  <snippet name="our_theta_360">
   <global></global>
   <function>return value%360</function>
   <linkedSource>/our_odom/pose/orientation/yaw_deg</linkedSource>
  </snippet>
  <snippet name="scout_theta_360">
   <global></global>
   <function>return value%360</function>
   <linkedSource>/real_time_od/pose/orientation/yaw_deg</linkedSource>
  </snippet>
  <snippet name="gt_theta_360">
   <global></global>
   <function>return value%360</function>
   <linkedSource>/real_time_gt/orientation/yaw_deg</linkedSource>
  </snippet>
 </customMathEquations>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

